### ISSUE TITLE
Separate unit tests and Quarkus tests into maven profiles

### ISSUE THREAD

**Issue by aptmac** [RedHatInsights/insights-runtimes-inventory#65]
As of https://github.com/RedHatInsights/insights-runtimes-inventory/commit/813f864bbd6fb7646d089f7c8ffb56f2d7ffd7a7 (and open PR https://github.com/RedHatInsights/insights-runtimes-inventory/pull/60), there are now "unit tests" that utilize the `@QuarkusTest` annotation and use TestContainers. The potential issue here is that TestContainers isn't friendly outside of Linux, and that the act of running a TestContainer may push the boundary between a unit and integration test.

It would be nice to separate these, so that running a simple `mvn package` could include the basic junit tests, and something like adding `-P integration` would run the integration tests.

**Comment by aptmac** [comment]
I took a look into this and came up with a wip (it's based on top of the open PR at the moment): https://github.com/aptmac/insights-runtimes-inventory/commit/f8695d8d7149b98ca68efa680c7b91c822fab800

I'll continue looking into this tomorrow, but there's some interesting cases here.

If a test class is annotated with `@QuarkusTest` then even if there's no test resource included it will start running some TestContainers. So I separated the "integration tests" into their own class, and kept the "unit tests" back without the annotation. So here we have `EventConsumerTests` and `EventConsumerIntegrationTest`.

I added a tag to the integration test classes, this can be used with the surefire plugin to include/exclude these tests based on a maven profile. The thing here is that we'd need two plugin declarations for the surefire plugin in the root pom xml; one for excluding the integration tests and one that doesn't exclude them.

Up until this point it isn't too tricky. But coverage makes it a bit less straightforward. I had included jacoco coverage as an optional tool behind a profile `-P coverage`. To keep this behind a profile is tougher because we'd need to have two separate versions of jacoco; one for `@QuarkusTest` runs and one without. The jacoco-quarkus extension won't run unless at least one Quarkus test is run, so the jacoco plugin used for only the unit-tests must be regular jacoco. At the moment I've just enabled coverage by default, but will start here tomorrow to see if there's a nice way of separating this out.
